renderdoc/os/win32/win32_shellext.cpp

```diff
@@ -362,18 +399,219 @@ struct RDCThumbnailProvider : public IThumbnailProvider, IInitializeWithStream
       return E_NOTIMPL;
     }
 
-    const byte *jpgbuf = m_Thumb.pixels;
     size_t thumblen = m_Thumb.len;
     uint32_t thumbwidth = m_Thumb.width, thumbheight = m_Thumb.height;
+    byte *thumbpixels = nullptr;
 
-    if(jpgbuf == NULL)
-      return E_NOTIMPL;
+    if(m_Thumb.format == FileType::JPG)
+    {
+      const byte *jpgbuf = m_Thumb.pixels;
+      if(jpgbuf == NULL)
+        return E_NOTIMPL;
+
+      int w = thumbwidth;
+      int h = thumbheight;
+      int comp = 3;
+      thumbpixels = jpgd::decompress_jpeg_image_from_memory(jpgbuf, (int)thumblen, &w, &h, &comp, 3);
+    }
+    else
+    {
+      const ResourceFormatType compType = m_ddsData.format.type;
```

Comment:
`compType` is a bit of a misleading name for this, since there is another member `compType` in the format descriptor which gives the component type - i.e. UNORM vs FLOAT vs UINT etc. This would be better named I think as `resourceType` or `formatType`, if you want to have something more specific than just `type`.

Reply:
Thank you, I was not aware of that. Changed to `resourceType`.